ReactCommon/butter/function.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <butter/butter.h>
+
+#if (defined(_WIN32) && __has_include(<functional/functor.h>))
```

Comment:
Using Mso looks weird here. I wonder if we can do it in a way that community could benefit from it too.
E.g.
```cpp
#ifdef BUTTER_FUNCTION_OVERRIDE_INCLUDE
#include BUTTER_FUNCTION_OVERRIDE_INCLUDE
```